plugins/woocommerce-blocks/tests/e2e/utils/create-dynamic-content.ts

```diff
@@ -74,3 +63,42 @@ export const createPostFromTemplate = async (
 
 	return createPost( requestUtils, payload );
 };
+
+export const updateTemplatesContent = async (
+	requestUtils: RequestUtils,
+	templateId: string,
+	templatePath: string,
+	data: unknown
+) => {
+	const templateContent = await readFile( templatePath, 'utf8' );
+	const content = Handlebars.compile( templateContent )( data );
+
+	const payload = {
+		content,
+	};
+
+	return requestUtils.rest( {
+		method: 'POST',
+		path: `/wp/v2/templates/${ templateId }`,
+		data: { ...payload },
+	} );
+};
+
+export const updateProductCatalogContent = async (
```

Comment:
I see where you coming from and that was my initial approach. [Here's the example ](https://github.com/woocommerce/woocommerce/blob/af4bbb9a0ae5ec913dbe082ac380a9e96ac11882/plugins/woocommerce-blocks/tests/e2e/tests/filter-blocks/attribute-filter.block_theme.spec.ts)of Attribute Filter tests at earlier stage of this PR.

It was exactly as you described but then I saw there's a common pattern and code repetition that I could lift to the util. In addition, I think there's a lot of scenarios based on Product Catalog template (the main template in the whole shop) and I agree - it's not generic but at the same time it isn't random one-time use template, hence I thought it may be ok to provide that more specific helper in here.

I'm ok with moving that helper to `tests/filter-blocks/utils.ts`. But at the same time, as we move more and more towards using `create-dynamic-content.ts` I believe there will be more scenarios based on Product Catalog template.

Let me know what you think about it?

Reply:
@kmanijak I can totally see where you're coming from.

Looking at this with fresh eyes and based on what you're saying I mostly agree. 

So now I better understand what you're doing I think the only thing that bothers me is the hard-coded `/shop` link. I wonder if we really need to `use` anything in this situation. Since a template and a page are sort of decoupled from each other, maybe we can just hardcode navigating to the page in the test and remove the API where we give a `link` via the test fixture? 

Additionally, would it be unreasonable to just call this `updateProductCatalogTemplate`?

Reply:
> So now I better understand what you're doing I think the only thing that bothers me is the hard-coded /shop link. I wonder if we really need to use anything in this situation. Since a template and a page are sort of decoupled from each other, maybe we can just hardcode navigating to the page in the test and remove the API where we give a link via the test fixture?

We could remove the link from `util` and hardcode it in the test but I still need to import the particular template instance in a test so the test uses a correct template version.

<img width="784" alt="image" src="https://github.com/woocommerce/woocommerce/assets/20098064/fa89e25b-b243-4420-981b-01abf344db5c">

For example in `attribute-filter.block_theme.spec.ts` we have three template specs: `defaultBlockTemplate`, `templateWithShowCounts`, `dropdownBlockTemplate`. With the template, the issue is they use the same instance hence I didn't find an alternative way to let the test know which "version" of the template it should use. With post it was easy, you use specific (and unique) post instances.

Reply:
@kmanijak I think thats why it doesn't make much sense to use fixtures here? You can set up the template once in the test? or if multiple tests use a template you can group them with decribe and use beforeAll?